src/Compatibility/Microsoft.DotNet.Compatibility.Shared/ValidatorBootstrap.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#nullable enable
+using System.Collections.Generic;
+using Microsoft.DotNet.ApiCompatibility.Logging;
+using Microsoft.DotNet.PackageValidation;
+using Microsoft.DotNet.PackageValidation.Validators;
+
+namespace Microsoft.DotNet.Compatibility
+{
+    internal static class ValidatorBootstrap
```

Comment:
I'm honestly not seeing the benefit of having this shared project over what we used to have. This code belongs to the layer that understands how to get a package and queue ApiCompat runs for each pair that should be compared (which means Microsoft.DotNet.PackageValidation) so  I don't think adding this extra layer is adding any value, and instead just adding complexity IMO

Reply:
I have to disagree here. While the ApiCompatibility layer offers the necessary API to compare left with right(s), the PackageValidation layer exposes the API to read from a package, create an internal representation of it (`Package` type) and to validate it (via the Validators). The frontend (console, msbuild, whatever) is responsible for parsing passed in arguments and passing them in to the exposed APIs. The frontend implements a logger (based on the platform) and decides which validators to use (based on the passed in arguments).

The `ValidatorBootstrap` class contains the shareable frontend code and does exactly the above:
1. Leverages the PackageValidation.Package API to create an internal representation of the passed in packages
2. Instantiates and invokes the desired validators based on the user input via the PackageValidation.Validators API.
3. Invokes the exposed ApiCompatibility.Logging API to generate the suppression file.

Reply:
Right, what you are saying is that it would be desired to be able to test one layer above and debug, so that could be done by adding scenario tests. Having a console project that isn't used at all (other than locally by someone) a) is prone to get broken as no one will execute it as part of the build, b) In this case it also complicates layering, c) it doesn't really add value for regression testing. If we instead were to add a scenario tests project that tested the whole layer and ran for each build, you would be adding extra protection to the code, while not having the protection be fragile and prone to break since it would be covered by CI.